### PR TITLE
[PAY-1137][PAY-1132] Infinite loading of albums on mobile favorites screen

### PR DESCRIPTION
**Pull request by schottra**
### Description
This change consolidates the paginated fetching logic between desktop/mobile as well as implementing the paginated experience on the mobile favorites screen.

* Updated hook for using fetched collections
  *  Automatic fetching of the first page of results whenever parameters change
  * Incorporation of the "is visible" logic from mobile to hide collections in certain cases
* Renamed selectors for saved collections to indicate they are `AccountCollection` items.
* Updated `CollectionList` to support loading data in leaf components
  * Added optional `collectionIds` prop which, when used, will trigger the new behavior
  * Added wrapper versions of the components at each level which will fetch the data and then use the original versions for layout
  * _Note_: I didn't expose new components, just new props which will trigger the new behavior. Long-term, we will want to migrate all usages of `CollectionList` to follow this pattern. But that's way out of scope and this will support both styles of usage.
* Removed computation of offline track count on mobile from the page-level hook. `CollectionCards` now calculate their own track count override based on offline status.
* Moved logic for offline track status to a hook and made the online computation skipping optional (so it's useful in cases where we want the count even when we _are_ online)
* Updated `AlbumsTab` and `PlaylistsTab` to use FlatList infinite scrolling behavior
* Made loading spinner while refreshing consistent across all three tabs on favorites screen
* Removed `isOfflineModeEnabled` checking in components I touched, since that flag is now fully shipped
* Moved filtering logic for mobile into a common place so it can be used on desktop
* Added documentation
* Cleanup of unused code
* Cleanup of a default export I introduced in my previous PR

### Dragons
I've verified as many corner cases as I could think of. But since this changes some complicated logic in the favorites screen, there is a chance I've broken some less-used flows.

### How Has This Been Tested?
Manual testing on iOS simulator and physical device.

### How will this change be monitored?
Regular release flow with testing on RC

### Feature Flags ###
N/A

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/pay-1137-infinite-load-albums-mobile

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/pay-1137-infinite-load-albums-mobile

**Review by piazzatron — APPROVED** [review]
Looks great!! Nice idea with the CollectionList migration pattern. Would appreciate a pair of client team eyes as well since this touches offline mode, but all looks good to me here.

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/pay-1137-infinite-load-albums-mobile

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/pay-1137-infinite-load-albums-mobile

File: packages/mobile/src/screens/favorites-screen/FavoritesScreen.tsx [hunk 1]
```diff
@@ -44,26 +36,18 @@ const favoritesScreens = [
...
       <ScreenHeader
         text={messages.header}
         icon={IconFavorite}
         styles={{ icon: { marginLeft: 3 } }}
       >
-        {isOfflineModeEnabled ? <FavoritesDownloadSection /> : null}
+        <FavoritesDownloadSection />
       </ScreenHeader>
-      <ScreenContent isOfflineCapable={isOfflineModeEnabled}>
+      <ScreenContent isOfflineCapable={true}>
```
Comment (dylanjeffers) [inline packages/mobile/src/screens/favorites-screen/FavoritesScreen.tsx:50]: nit, but we do `<ScreenContent isOfflineCapable>`

File: packages/mobile/src/screens/favorites-screen/PlaylistsTab.tsx [hunk 2]
```diff
@@ -39,15 +40,31 @@ export const PlaylistsTab = () => {
...
   const [filterValue, setFilterValue] = useState('')
-  const { filteredCollections: userPlaylists, collectionIdsToNumTracks } =
-    useCollectionScreenData(filterValue, 'playlists')
-  const isOfflineModeEnabled = useIsOfflineModeEnabled()
+  const {
+    collectionIds: userPlaylists,
+    hasMore,
+    fetchMore,
+    status
+  } = useCollectionScreenData({
```
Comment (dylanjeffers) [inline packages/mobile/src/screens/favorites-screen/PlaylistsTab.tsx:48]: `useCollections` maybe? but all good

File: packages/mobile/src/screens/favorites-screen/PlaylistsTab.tsx [hunk 3]
```diff
@@ -39,15 +40,31 @@ export const PlaylistsTab = () => {
...
+    }
+  }, [isReachable, hasMore, fetchMore])
+
+  const loadingSpinner = <LoadingMoreSpinner />
+
   return (
     <VirtualizedScrollView>
-      {!userPlaylists?.length && !filterValue ? (
-        isOfflineModeEnabled && !isReachable ? (
+      {!statusIsNotFinalized(status) &&
```
Comment (dylanjeffers) [inline packages/mobile/src/screens/favorites-screen/PlaylistsTab.tsx:64]: is this worth breaking into constant?

File: packages/mobile/src/screens/favorites-screen/useCollectionScreenData.ts [hunk 4]
```diff
@@ -81,29 +107,18 @@ export const useCollectionScreenData = (
...
-      offlineTracksStatus || {}
-    )
-  }, [
-    isOfflineModeEnabled,
-    isReachable,
-    offlineTracksStatus,
-    filteredCollections
-  ])
 
   return {
```
Comment (dylanjeffers) [inline packages/mobile/src/screens/favorites-screen/useCollectionScreenData.ts:118]: wow so complex, but all is written so clean

**Review by dylanjeffers — APPROVED** [review]
what a momentous change, high quality work. excited to test next week for release.

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/pay-1137-infinite-load-albums-mobile

**Pull request merged**